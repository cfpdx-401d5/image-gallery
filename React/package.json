{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
