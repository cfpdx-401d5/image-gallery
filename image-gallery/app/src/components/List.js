import React from 'react';

export default function List(props) {
    const listItem = props.puppies.map(puppies => {
        return (
            <ListDisplay key={puppies._id} puppies={puppies} onDelete={props.onDelete}/>
        );
    });
    return (
      <ul>{listItem}</ul>
    );
}

function ListDisplay(props) {
    return (
        <div className="clearfix">
          <li>
            <p>Title: {props.puppies.title}</p>
            <p>Description: {props.puppies.description}</p>
            <p>Url: <a href={props.puppies.url}>{props.puppies.url}</a></p>
            <div className="delete"><button className="shadow animate orange" onClick={() => props.onDelete(props.puppies._id)}>Delete</button></div>
          </li>
        </div>
    );
}

