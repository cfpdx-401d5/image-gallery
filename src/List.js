import React, { PropTypes } from 'react';

function List(props) {
  return (
    <ul>
      <li><a href={props.image.url}>{props.image.title}</a>: {props.image.description}
      </li>
    </ul>
  );
}
List.propTypes = {
  image: PropTypes.shape({
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,
};

export default List;
