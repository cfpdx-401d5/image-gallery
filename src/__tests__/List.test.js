/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import List from '../List';

describe('List component', function () {
    it('renders the List component', () => {
        const rendered = shallow(<List image={{url:'_url', title: '_title', description: 'desc'}}/>);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});