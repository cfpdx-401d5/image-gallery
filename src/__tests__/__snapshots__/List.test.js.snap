exports[`List component renders the List component 1`] = `
<ul>
  <li>
    <a
      href="_url">
      _title
    </a>
    : 
    desc
  </li>
</ul>
`;
